src/lib/QueryType/UserQueryType.php

```diff
@@ -0,0 +1,110 @@
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\AdminUi\QueryType;
+
+use Ibexa\Contracts\Core\Repository\Values\Content\Query;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Core\QueryType\BuiltIn\AbstractQueryType;
+use RuntimeException;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+final class UserQueryType extends AbstractQueryType
+{
+    private const USER_ADMIN_ID = 14;
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        parent::configureOptions($resolver);
+
+        $resolver->setDefined('phrase');
+        $resolver->setAllowedTypes('phrase', ['string', 'null']);
+
+        $resolver->setDefined('extra_criteria');
+        $resolver->setAllowedTypes('extra_criteria', [Criterion::class, 'null']);
+
+        $resolver->setDefaults(
+            [
+                'phrase' => null,
+                'extra_criteria' => null,
+            ]
+        );
+    }
+
+    /**
+     * @param array<mixed> $parameters
+     */
+    protected function doGetQuery(array $parameters): Query
+    {
+        $parameters['filter']['siteaccess_aware'] = false;
+
+        return parent::doGetQuery($parameters);
+    }
+
+    /**
+     * @param array<mixed> $parameters
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidCriterionArgumentException
+     */
+    protected function getQueryFilter(array $parameters): Criterion
+    {
+        $criteria = [
+            new Criterion\IsUserEnabled(),
+            $this->excludeSystemUsers(),
+        ];
+
+        if (!empty($parameters['extra_criteria'])) {
+            $criteria[] = $parameters['extra_criteria'];
+        }
+
+        if (!empty($parameters['phrase'])) {
+            $phrase = $this->cleanSearchPhrase($parameters['phrase']);
+            $criteria[] = new Criterion\LogicalOr(
+                [
+                    new Criterion\ContentName('*' . $phrase . '*'),
+                    // Used with EQ operator and without wildcards due to hashing email in solr and elasticsearch
+                    new Criterion\UserEmail($phrase, Criterion\Operator::EQ),
+                ]
+            );
+        }
+
+        return new Criterion\LogicalAnd($criteria);
+    }
+
+    public static function getName(): string
+    {
+        return 'IbexaAdminUi:User';
+    }
+
+    private function cleanSearchPhrase(string $phrase): string
+    {
+        $sanitizedPhrase = preg_replace('/[^a-zA-Z0-9@._-]/', '', $phrase);
+        if (null === $sanitizedPhrase) {
+            throw new RuntimeException('Could not to sanitize search phrase.');
+        }
+
+        return $sanitizedPhrase;
+    }
+
+    private function excludeSystemUsers(): Criterion
+    {
+        return new Criterion\LogicalNot(
+            new Criterion\ContentId(
+                [
+                    self::USER_ADMIN_ID,
```

Comment:
Shouldn't this be applied to all the users with administrative permissions?

Reply:
We can try to filter out all users under Administrator and Anonymous using SubtreeCriterion

Reply:
Why not just taking administrator user group into account?

Reply:
The administrative group seems obvious, but I wondering if searching anonyous users will be useful.

Reply:
Fixed: 
https://github.com/ibexa/admin-ui/pull/1372/commits/9428b5a6297f7ea4676dbd81ddb4dd404de09c11
https://github.com/ibexa/admin-ui/pull/1372/commits/57c7735301b02234d8afdbb4f03467b06ca97bf8